app/assets/javascripts/components/students/components/Articles/SelectedStudent/RevisionsList/StudentRevisionsList.jsx

```diff
@@ -8,15 +8,14 @@ import StudentRevisionRow from './StudentRevisionRow';
 
 // Libraries
 import CourseUtils from '~/app/assets/javascripts/utils/course_utils.js';
-import ArticleUtils from '~/app/assets/javascripts/utils/article_utils.js';
 import studentListKeys from '@components/students/shared/StudentList/student_list_keys.js';
 
 export class StudentRevisionsList extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       isOpen: false,
-      namespace: 'all'
+      namespace: -1 // show all namespace revisions by default
```

Comment:
For a code comprehensibility perspective, I don't think replacing strings with numbers here is helpful. Do the changes here have any effect on behavior? If they are unrelated to displaying stats on the Overview tab, they probably shouldn't be included in this PR.

Reply:
These changes are related to the namespace filter of revisions table.
Actually, earlier this code used `namespaceToId` function from ArticleUtils. But, now that I have defined new mapping function there (`NamespaceIdMapping`), that function would have become its duplicate. Hence, some changes reflected here.
Should I open another pr then?

Reply:
I think this is okay, except... is there any reason to use `-1` in place of `all`? The other integer values all map to a mainspace, but this one does not and `-1` not as easy to interpret as `all`.

Reply:
Actually, I don't know what number to use for 'all', because for other options it would be their namespace id. May be, we can still use the strings - 'all', 'article', 'talk', etc.

Reply:
I think a mix of numbers (for actual namespaces) and a string for `all` would be okay. But also, using strings for all of the values seems find to me.